hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientGrpc.java

```diff
@@ -339,6 +343,11 @@ private XceiverClientReply sendCommandWithRetry(
         // in case these don't exist for the specific datanode.
         reply.addDatanode(dn);
         responseProto = sendCommandAsync(request, dn).getResponse().get();
+        if (LOG.isDebugEnabled()) {
```

Comment:
In sendCommandAsync() below we already have a LOG.debug(cmdType, Dn). Can you move this there to avoid duplicate debug log?

Reply:
@xiaoyuyao I hope so, but I think it would be difficult, because, i want to log the retry index and cost time.
when the `sendCommandAsync ` return back, the client still doesn't get the result from datanode. the result return until the `get` return.

If i have something wrong, please correct me.

Reply:
I mean if we could move the LOG for latency around Line 464 where the container op latency metrics are updated. This way, we can piggyback LOG with the existing metrics update. 

metrics.addContainerOpsLatency(request.getCmdType(),
                    System.nanoTime() - requestTime);

Reply:
@xiaoyuyao Thanks you for your reply and suggestion. I tried to add the LOG into the L464, like following
```java
              public void onNext(ContainerCommandResponseProto value) {
                replyFuture.complete(value);
                metrics.decrPendingContainerOpsMetrics(request.getCmdType());
                long cost = System.nanoTime() - requestTime;
                metrics.addContainerOpsLatency(request.getCmdType(),
                    cost);
                if (LOG.isDebugEnabled()) {
                  LOG.debug("Executed command {} on datanode {}, cost = {}, "
                          + "retryCount = {}" + ", cmdType = {}", request, dn,
                      cost, index, request.getCmdType());
                }
                semaphore.release();
              }
```

But, there are some problem
- It cannot print the `Index` which stand for the retried datanode index of the `datanodeList`.
- I should add the same log logic to the `onError` also? Otherwise, when we met error, the `onNext` won't be executed.

As `index` can be calculate from the log context, I drop it.